instrumentation/kafka-streams/src/main/java/brave/kafka/streams/KafkaStreamsTracing.java

```diff
@@ -244,6 +247,10 @@ public KafkaStreams kafkaStreams(Topology topology, Properties streamsConfig) {
 
   /**
    * Create a filter transformer.
+   *
+   * WARNING: as this filter is based on Transformer API, if followed by a grouping or joining
```

Comment:
Maybe a bit more elaborate like

> WARNING: this filter implementation uses the Streams <a href="https://kafka.apache.org/22/javadoc/org/apache/kafka/streams/kstream/KStream.html#transform-org.apache.kafka.streams.kstream.TransformerSupplier-java.lang.String...-}">transform</a> API, meaning that re-partitioning can occur if a key modifying operation like grouping or joining operation is applied after this filter. In that case, consider using {link #markFilter} instead which uses <a href="https://kafka.apache.org/22/javadoc/org/apache/kafka/streams/kstream/KStream.html#transformValues-org.apache.kafka.streams.kstream.ValueTransformerSupplier-java.lang.String...-">transformValues</a> API instead.


(javadoc links might be not correct but you get the idea)